test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs

```diff
@@ -1674,6 +1698,26 @@ public void DifferentCultureJsonWebToken()
             Assert.Equal("12.2", numericList[0].Value);
             Assert.Equal("11.1", numericList[1].Value);
         }
+
+        // Test to verify equality between JsonWebTokens created from a string and an equivalent span
+        [Theory, MemberData(nameof(ParseTokenTheoryData))]
+        public void CompareJsonWebToken(JwtTheoryData theoryData)
```

Comment:
nit: clarifies what the test does and the expected result.
```suggestion
        public void StringAndMemoryConstructors_CreateEquivalentTokens(JwtTheoryData theoryData)
```